### PR TITLE
1.3.0 changelog and release notes

### LINKED ISSUE #1103: Create Data Prepper 1.3.0 Release Notes
Create Data Prepper 1.3 Release Notes

All changes should be available at:

https://github.com/opensearch-project/data-prepper/milestone/2?closed=1

The release notes should provide a high-level overview of the changes we made to Data Prepper. They should include at least:

* New Features
* Enhancements
* Bug Fixes
* Migrations - Any migrations customers should make
* Breaking Changes - Any breaking changes, even if minor.

### LINKED ISSUE #1195: Create Data Prepper 1.3.0 Changelog
Create Data Prepper 1.3 Changelog

The Changelog is a detailed overview of all the changes made to Data Prepper in this release. It needs to be generated from Git history.

### PR DESCRIPTION
**Pull request by chenqi0805**
Signed-off-by: Chen <19492223+chenqi0805@users.noreply.github.com>

### Description
This PR adds for 1.3.0 release

* changelog generated by git-release-notes 1.2.1..HEAD markdown
* manually picked release notes
 
### Issues Resolved
Resolves #1195 .
Resolves #1103 .
 
### Check List
- [ ] New functionality includes testing.
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: release/release-notes/data-prepper.release-notes-1.3.0.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+* Upgrade docker-compose.yml files from ODFE to OpenSearch ([#847](https://github.com/opensearch-project/data-prepper/pull/847))
+* Update the ADOT example to OpenSearch ([#703](https://github.com/opensearch-project/data-prepper/pull/703))
+* Migrate demo sample application to use opensearch and opensearch dashboards instead of ODFE and kibana ([#666](https://github.com/opensearch-project/data-prepper/pull/666))
+* Update log ingestion example to use latest data-prepper docker image ([#752](https://github.com/opensearch-project/data-prepper/pull/752))
+
+### Refactoring
+* Deprecating Record and RecordMetaData ([#954](https://github.com/opensearch-project/data-prepper/pull/954))
+* FileSink to use Event model ([#750](https://github.com/opensearch-project/data-prepper/pull/750))
+* StringPrepper to use Event model ([#753](https://github.com/opensearch-project/data-prepper/pull/753))
+* Rename Prepper to Processor ([#655](https://github.com/opensearch-project/data-prepper/pull/655), [#667](https://github.com/opensearch-project/data-prepper/pull/667))
```
Comment (dlvenable) [inline release/release-notes/data-prepper.release-notes-1.3.0.md:57]: I'd like to make a call-to-action somewhere in the release notes. I'd like to let users know that along with this change, we have updated the Pipeline configuration to support `processor:` and that we will be removing `prepper:` in 2.0. I think clearly noting this deprecation is important.

I'm open to different ways of doing it, but maybe have a section higher up for deprecations?

```
### Deprecations
```

**Review by dlvenable — CHANGES_REQUESTED** [review]

File: release/release-notes/data-prepper.release-notes-1.3.0.md (hunk 1 above)
Comment (chenqi0805) [reply release/release-notes/data-prepper.release-notes-1.3.0.md:57]: I will update the description and move it to a Deprecation section

**Review by dlvenable — APPROVED** [review]

File: release/release-notes/data-prepper.release-notes-1.3.0.md [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+* DropProcessor to filter out (remove/drop) entire events ([#801](https://github.com/opensearch-project/data-prepper/pull/801))
+
+### Enhancements
+* Add Data Prepper expression evaluator ([#1024](https://github.com/opensearch-project/data-prepper/pull/1024), [#1027](https://github.com/opensearch-project/data-prepper/pull/1027), [#1090](https://github.com/opensearch-project/data-prepper/pull/1090), [#1065](https://github.com/opensearch-project/data-prepper/pull/1065), [#1153](https://github.com/opensearch-project/data-prepper/pull/1153), [#1155](https://github.com/opensearch-project/data-prepper/pull/1155), [#1169](https://github.com/opensearch-project/data-prepper/pull/1169), [#1177](https://github.com/opensearch-project/data-prepper/pull/1177), [#1178](https://github.com/opensearch-project/data-prepper/pull/1178))
+* Add conditional to DropEventsProcessor ([#1174](https://github.com/opensearch-project/data-prepper/pull/1174))
+* Support for Data Prepper expression in Data Prepper core ([#1157](https://github.com/opensearch-project/data-prepper/pull/1157))
```
Comment (sbayer55) [inline release/release-notes/data-prepper.release-notes-1.3.0.md:18]: nit: This line is redundant with [line 16](https://github.com/opensearch-project/data-prepper/pull/1201/files#diff-46cb9eccea41269b336de54f506849fb80fe6a1a13d643bb1eb2a7f479d46309R16)

File: release/release-notes/data-prepper.release-notes-1.3.0.md [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+### Features
+* AggregateProcessor for generic stateful aggregation ([#839](https://github.com/opensearch-project/data-prepper/pull/839), [#850](https://github.com/opensearch-project/data-prepper/pull/850), [#931](https://github.com/opensearch-project/data-prepper/pull/931), [#969](https://github.com/opensearch-project/data-prepper/pull/969), [#1022](https://github.com/opensearch-project/data-prepper/pull/1022), [#1046](https://github.com/opensearch-project/data-prepper/pull/1046))
+* DateProcessor to extract dates from fields in events ([#971](https://github.com/opensearch-project/data-prepper/pull/971), [#1014](https://github.com/opensearch-project/data-prepper/pull/1014))
+* Processors to support mutate, alter, and delete fields from Events ([#1002](https://github.com/opensearch-project/data-prepper/pull/1002))
+* KeyValueProcessor to support parsing messages with key-value strings such as queries and properties ([#872](https://github.com/opensearch-project/data-prepper/pull/872))
+* DropProcessor to filter out (remove/drop) entire events ([#801](https://github.com/opensearch-project/data-prepper/pull/801))
```
Comment (sbayer55) [inline release/release-notes/data-prepper.release-notes-1.3.0.md:13]: Consider expanding on this to include conditional filtering/logic.

**Review by sbayer55 — DISMISSED** [review]
Thanks for writing this @chenqi0805. Your descriptions are concise, informative and digestible

**Review by dlvenable — APPROVED** [review]
Nice improvements! Thanks for the README suggestion @sbayer55 !

**Review by sbayer55 — APPROVED** [review]

**Pull request merged**